src/snabbdom.ts

```diff
@@ -1,55 +1,90 @@
-/* global module, document, Node */
-import {Module} from './modules/module';
-import {Hooks} from './hooks';
-import vnode, {VNode, VNodeData, Key} from './vnode';
-import * as is from './is';
-import htmlDomApi, {DOMAPI} from './htmldomapi';
-
-function isUndef(s: any): boolean { return s === undefined; }
-function isDef(s: any): boolean { return s !== undefined; }
+/* global Node */
+import {Hooks, InsertHook, VNode, VNodes, Key, VText, VComment,
+        Module, VElement, VFragment, DOMAPI, PreHook, CreateHook, UpdateHook, DestroyHook, RemoveHook, PostHook} from './types';
 
-type VNodeQueue = Array<VNode>;
+const { isArray } = Array;
 
-const emptyNode = vnode('', {}, [], undefined, undefined);
+const ELEMENT_NODE = 1, TEXT_NODE = 3, COMMENT_NODE = 8, DOCUMENT_FRAGMENT_NODE = 11;
```

Comment:
I think introducing the `nt` property on nodes is a great idea. But instead of these constants we could use a const enum:

```ts
const enum VNodeType { Empty, Element, Text, Comment, Fragment };
```

Const enums add no run-time overhead (they're compiled to number literals) but they provide extra type-safety and look nicer.

If using the same numbers as the DOM is important then that can also be done with const enums.

```ts
const enum VNodeType {
  Empty = 0,
  Element = 1,
  Text = 3,
  ...
}
```

Reply:
will do!